### PR TITLE
[Runtime] Add dump() methods to TypeContextDescriptor and EnumDescriptor.

### PR DESCRIPTION
**Pull request by mikeash**

### CODE REVIEW

**Comment by mikeash** [comment]
@swift-ci please smoke test

File: stdlib/public/runtime/Metadata.cpp [hunk 1]
```diff
@@ -3945,6 +3949,33 @@ void Metadata::dump() const {
   printf("Type Context Description: %p.\n", getTypeContextDescriptor());
   printf("Generic Args: %p.\n", getGenericArgs());
 }
+
+template <>
+LLVM_ATTRIBUTE_USED
+void TypeContextDescriptor::dump() const {
+  printf("TargetTypeContextDescriptor.\n");
```
Comment (gottesmm) [inline stdlib/public/runtime/Metadata.cpp]: Looks awesome! My only question is why you are using this in only some of the places.

Comment (mikeash) [reply stdlib/public/runtime/Metadata.cpp]: I banged these out while debugging a problem and figured I should add them in. We should definitely make more though.

Comment (gottesmm) [reply stdlib/public/runtime/Metadata.cpp]: No that isn't what I meant. I meant in some parts you are using "this->" and others you aren't.

Comment (mikeash) [reply stdlib/public/runtime/Metadata.cpp]: Oh, `this`! Hah. The compiler complained about not finding those fields without it. I don't really understand why.

Comment (gottesmm) [reply stdlib/public/runtime/Metadata.cpp]: From talking with JoeG it is due to dependent member lookup and c++ being what it is. This looks great to me!

**Review by gottesmm — APPROVED** [review]

File: stdlib/public/runtime/Metadata.cpp (hunk 1 above)
Comment (mikeash) [reply stdlib/public/runtime/Metadata.cpp]: Thanks!

**Pull request merged**